test/Microsoft.AspNetCore.JsonPatch.Test/IntegrationTests/ExpandoObjectIntegrationTest.cs

```diff
@@ -0,0 +1,841 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Dynamic;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNetCore.JsonPatch
+{
+    public class ExpandoObjectIntegrationTest
+    {
+        [Fact]
+        public void AddNewProperty_ToExpandoObject()
+        {
+            dynamic obj = new ExpandoObject();
+            obj.Test = 1;
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("NewInt", 1);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(obj);
+
+            Assert.Equal(1, obj.NewInt);
+            Assert.Equal(1, obj.Test);
+        }
+
+        [Fact]
+        public void AddNewProperty_ToExpandoOject_InTypedObject()
```

Comment:
I think that there are way too many of these kinds of tests. Putting the target object at another level of testing doesn't make it a valuable test. A few of these make sense.